lib/liquid/tags/render.rb

```diff
@@ -47,17 +47,19 @@ def render_tag(context, output)
       context_variable_name = @alias_name || template_name.split('/').last
 
       render_partial_func = ->(var, forloop) {
-        inner_context               = context.new_isolated_subcontext
-        inner_context.template_name = template_name
-        inner_context.partial       = true
-        inner_context['forloop']    = forloop if forloop
-
-        @attributes.each do |key, value|
-          inner_context[key] = context.evaluate(value)
+        partial.with_context_restrictions(context) do
```

Comment:
This will allow template files to apply certain restrictions to rendering if needed

Reply:
Could this be done by overriding `render_to_output_buffer` instead?

Reply:
Overriding `render_to_output_buffer` would not be enough unfortunately because attributes that are passed in to the partial are evaluated here, outside of the partial's rendering (line 56 below). Overriding `render_to_output_buffer` would not restrict what's passed in as argument to render and therefore we would not be able to block restrict cases like below:

```liquid
{% assign t = theme %}
{% render "snippet_name", theme: t %}
```

Reply:
By doing it at this level we avoid having to deal with restricting what's passed into attributes below at line 56.

Reply:
Actually, I take it back, overriding `render_to_output_buffer` is a much cleaner approach. I'll revert the introduction of `with_context_restrictions`